File: day6.py
# day 6
from collections import deque

def solve(n):
    with open('data/day6.txt', 'r') as f:
        data = f.read()

    window = deque(maxlen=n)
    for i in range(len(data)):
        window.append(data[i])
        if len(set(window)) == n:
            return i + 1


if __name__ == "__main__":
    print(f"Part 1: {solve(4)}")
    print(f"Part 2: {solve(14)}")

File: day6.clj
(ns day6)

(def input-data "/Users/tom/Dev/python/advent2022/data/day6.txt")

(defn solve [s n]
   (+ n (count (take-while #(< (count %) n)
                           (map set (partition n 1 (seq s)))))))

(comment
  ;; Part 1
  (solve (slurp input-data) 4)

  ;; Part 2
  (solve (slurp input-data) 14))
